src/core/service.c

```diff
@@ -3391,6 +3391,11 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
         else
                 clean_mode = EXIT_CLEAN_DAEMON;
 
+        if (s->main_pid == pid)
```

Comment:
If there are still processes active in the cgroup after main_pid has exited (`ExitType=cgroup`), would this cause issues? Or is it unrelated?

Reply:
I think that's fine. By closing this even source, we are saying that we are not interested in the exec-fd event. But we know that this particular process has dies already, so we don't need the information from exec-fd anymore. Whether there are other process or not doesn't seem relevant.